### ISSUE TITLE
Invalid combinations of --out, --outDir, and --rootDir should cause a compiler error.

### ISSUE THREAD

**Issue by paulvanbrenk** [Microsoft/TypeScript#3305]
When using invalid combinations of ```--out```, ```--outDir```, and ```--rootDir```, the compiler currently doesn't report any errors, but emits files in 'random' locations. Further when you specify ```--rootDir``` without ```--outDir``` this is ignored as well.

**Expected** to get errors/warnings from the compiler in all those cases.
